client/consul/consul.go

```diff
@@ -29,3 +36,101 @@ type ServiceIdentityAPI interface {
 type SupportedProxiesAPI interface {
 	Proxies() (map[string][]string, error)
 }
+
+// JWTLoginRequest is an object representing a login request with JWT
+type JWTLoginRequest struct {
+	JWT            string
+	Role           string
+	AuthMethodName string
+}
+
+// ConsulClient is the interface that the nomad client uses to interact with
+// Consul.
+type ConsulClient interface {
+	// DeriveSITokenWithJWT contacts the nomad server and fetches wrapped tokens
+	// for a set of tasks. The wrapped tokens will be unwrapped using consul and
+	// returned.
```

Comment:
I think there's some copypaste from the existing code in this comment? In this method we have the JWT already so we're contacting Consul directly, not the Nomad server. And I'm not sure what "unwrapped using consul" means here

Reply:
This and other copypasta errors are my bad—I went back and forth a few times on this PR because I wasn't sure how to handle token renewals/re-issues, and there's some leftover mess. Fixed in [4c925f3](https://github.com/hashicorp/nomad/pull/18370/commits/4c925f3a61b8b612482abe61fc7013dab06fd041)